src/Client/Connection.cpp

```diff
@@ -288,6 +309,22 @@ void Connection::forceConnected(const ConnectionTimeouts & timeouts)
     }
 }
 
+#if USE_SSL
+void Connection::sendCluster()
+{
+    salt.clear();
+    pcg64_fast rng(randomSeed());
+    UInt64 rand = rng();
+    std::string_view data(reinterpret_cast<char *>(&rand), sizeof(rand));
```

Comment:
Minor: Maybe the code would be clearer if we had an overload of `encodeSHA256` like this:

```
void encodeSHA256(const void * data, size_t size, unsigned char * out);
```

so we didn't have to use this `std::string_view data`. And a comment here about why you call `salt.resize(32)` would be nice.

Reply:
> I think the code would be clearer if we had an overload of encodeSHA256 like this:
>void encodeSHA256(const void * data, size_t size, unsigned char * out);

I though about this, but was going to make it separately.
Ok, added (rebasing and keep only final patches and make them clean will be a nightmare)

> And a comment here about why you call salt.resize(32) would be nice.

There are lot's of other places, that don't have it, hence I decided that it is ok (although I was a bit surprised to see magic number 32, but I guess everybody knows that it is sha256):

```c
programs/install/Install.cpp-                std::vector<uint8_t> hash;
programs/install/Install.cpp:                hash.resize(32);
programs/install/Install.cpp-                encodeSHA256(password, hash.data());
--
src/Access/Authentication.h-    Digest hash;
src/Access/Authentication.h:    hash.resize(32);
src/Access/Authentication.h-    ::DB::encodeSHA256(text, hash.data());
--
src/Storages/MergeTree/MergeTreePartition.cpp-    hash.get128(hash_data);
src/Storages/MergeTree/MergeTreePartition.cpp:    result.resize(32);
src/Storages/MergeTree/MergeTreePartition.cpp-    for (size_t i = 0; i < 16; ++i)
```